### PR TITLE
Feat: Add SignUp Flow

### PR DESCRIPTION
**Pull request by abhishek310001**
## Description
_This PR aims to add the functionality of SignUp to the Boilerplate code.
The UI can be seen in the screenshots and functionality in the video attached._

## Screenshots:

| Screenshot 2024-03-22 at 00 11 51 | Screenshot 2024-03-22 at 00 12 44 |
| ---- | ---- |
| Screenshot 2024-03-22 at 00 12 37 | Screenshot 2024-03-22 at 00 12 24 |

## Video:
_Video demonstrating feature of `Sign Up` flow._

https://github.com/jalantechnologies/rflask-boilerplate/assets/128154648/4e73da3d-826b-4d53-82a2-b77b71fd8e4d

_Video demonstrating all the validations of `Sign Up` page._

https://github.com/jalantechnologies/rflask-boilerplate/assets/128154648/30b2006c-b2e4-413c-bbd7-f7036afa3e6a

## Database schema changes

```
Account:
  id: str
  first_name: str
  hashed_password: str
  last_name: str
  username: str
```

### Automated test cases added
- Modified existing Automated test cases to match DB Schema changes

### Manual test cases run
_For each manual test case, list the steps to test or reproduce the PR._

- Verified on clicking `Sign Up` button with one or more of the field/s blank (First Name, Last Name, Email, Password, Confirm Password) didnot submit the form and raises error.
- Verified validations of all Input Fields
- Verified Account Creation on Clicking `Sign Up` button when all the information is correct.
- Checked raising of error if the same Email is already created.

### CODE REVIEW

File: src/apps/backend/modules/account/account_service.py [hunk 1]
```diff
@@ -24,6 +26,8 @@ def get_account_by_username_password(*, params: AccountSearchParams) -> Account:
     )
     return Account(
       id=str(account.id),
+      first_name=account.first_name,
       hashed_password=account.hashed_password,
```
Comment (jjalan) [inline src/apps/backend/modules/account/account_service.py:30]: I understand this is not related to this PR but`hashed_password` should not be populated on Account. Can we please remove it?

File: src/apps/frontend/services/account.service.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ export default class AccountService extends APIService {
       localStorage.getItem('access-token'),
     ) as AccessToken;
 
-    return this.apiClient.get(`/accounts/${userAccessToken.accountId}`, {
+    return this.apiClient.get(`/account/${userAccessToken.accountId}`, {
```
Comment (jjalan) [inline src/apps/frontend/services/account.service.ts:12]: Resource should always be plural in URI, please revert this change

File: src/apps/frontend/services/auth.service.ts [hunk 3]
```diff
@@ -3,11 +3,25 @@ import { AccessToken, ApiResponse } from '../types';
...
+
   login = async (
     username: string,
     password: string,
-  ): Promise<ApiResponse<AccessToken>> => this.apiClient.post('/access-tokens', {
-    username,
-    password,
-  });
+  ): Promise<ApiResponse<AccessToken>> =>
+    this.apiClient.post('/access-tokens', {
```
Comment (jjalan) [inline src/apps/frontend/services/auth.service.ts:23]: This is incorrect.

We need to probably create a Javascript object by parsing JSON object returned by api client.

File: src/apps/backend/modules/account/account_service.py (hunk 1 above)
Comment (abhishek310001) [reply src/apps/backend/modules/account/account_service.py]: Removed the `hashed_password` from Account class

File: src/apps/frontend/services/account.service.ts (hunk 2 above)
Comment (abhishek310001) [reply src/apps/frontend/services/account.service.ts]: Reverted and changed the endpoint from `account` to `accounts`

File: src/apps/frontend/services/auth.service.ts (hunk 3 above)
Comment (abhishek310001) [reply src/apps/frontend/services/auth.service.ts]: Done

**Review by jjalan — APPROVED** [review]
